src/main/webapp/js/query.js

```diff
@@ -0,0 +1,219 @@
+                throw 'No results';
+            else if (response.status === "NO_REROLLS")
+                throw 'No re-rolls left';
+            else
+                throw 'Unforeseen error';
+        })
+        .catch((error) => {
+            errorEl.classList.remove('success-banner');
+            errorEl.classList.remove('hidden');
+            errorEl.classList.add('error-banner');
+            errorEl.innerText = error;
+        });
+}
+
+$('#randomize-form').submit(function(event) {
+    const errorEl = document.getElementById("error");
+    errorEl.classList.add('hidden');
+
+    event.preventDefault();
+    let url = $(this).attr('action');
+    let lat = localStorage.getItem("lat");
+    let lng = localStorage.getItem("lng");
+    let queryStr = $(this).serialize() + `&lat=${lat}&lng=${lng}`;
+
+    $.ajax({
+        type: "POST",
+        url: url,
+        data: queryStr,
+        success: function(response) {
+            query();
+        }
+    });
+});
+
+$("input, textarea").blur(function() {
+    if ($(this).val() != "") {
+        $(this).addClass("active");
+    } else {
+        $(this).removeClass("active");
+    }
+});
+
+function resultsPage(pick) {
+    fetch(`../results.html`)
+        .then(html => html.text())
+        .then((html) => {
+            document.getElementById("page-container").innerHTML = html;
+            document.getElementById("pick").innerText = pick;
+        });
+}
+
+function reroll() {
+    const pickEl = document.getElementById("pick");
+    fetch(`/query`, { method: 'GET' })
+        .then(response => response.json())
+        .then((response) => {
+            responseJson = response; // Debug
+            if (response.status === "OK") {
+                pickEl.innerText = response.pick;
+            } else if (response.status === "INVALID_REQUEST")
+                throw 'Invalid request';
+            else if (response.status === "ZERO_RESULTS")
+                throw 'No results';
+            else if (response.status === "NO_REROLLS")
+                throw 'No re-rolls left';
+            else
+                throw 'Unforeseen error';
+        })
+        .catch((error) => {
+            pickEl.innerText = error;
+        });
+}
+
+/*
+    FUNCTIONS TO GET THE USER'S LOCATION AND TRANSFORM TO AN ADDRESS
+ */
+function getLocation() {
+    let location = document.getElementById("location-container");
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function(position) {
+            let pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
+            };
+            localStorage.setItem("lat", pos.lat);
+            localStorage.setItem("lng", pos.lng);
+            convertLocation(pos).then((address)=>{
+                console.log(address);
+                location.innerText = address;
+            });
+        });
+    } else {
+    // Browser doesn't support Geolocation
```

Comment:
indent

the indentation seems pretty inconsistent overall; could you run the formatter?

Reply:
formatted with prettier